ast/preload.go

```diff
@@ -0,0 +1,558 @@
+package ast
+
+import "strconv"
+
+// NOTE! Great care must be taken **not** to modify the terms returned
+// from these functions, as they are shared across all callers.
+
+var (
+	booleanTrueTerm  = &Term{Value: Boolean(true)}
+	booleanFalseTerm = &Term{Value: Boolean(false)}
+
+	// since this is by far the most common negative number
+	minusOneNumber = Number("-1")
+	minusOneTerm   = &Term{Value: minusOneNumber}
+)
+
+func CachedBooleanTerm(b bool) *Term {
```

Comment:
💭 Why not change `BooleanTerm`? I'm not sure I see the need to ever *not* use this one.